### DISCUSSION TITLE
LDAP Debugging

### DISCUSSION THREAD

**Discussion by zinsmeik** [TandoorRecipes/recipes#1225] [General]
Hi,

how can I debug LDAP Settings/Connection. I set up LDAP in .env but it is not working and I do not see any error or logging. Is there a way to debug LDAP connection maybe to a special file?
Thanks

**Comment by vabene1111** [comment]
i think you could set `DEBUG=1` in the environment file, i dont think there is any more debug than that. Other than that you might be able to get help from the person who added ldap support in the first place see #960 or in the offical documentation of the ldap library we are using https://www.python-ldap.org/en/python-ldap-3.3.0/

I myself sadly do now know very much about ldap so not sure how i can help but if there needs to be some flag in the application let me know

**Reply by zinsmeik** [reply]
Thank you @vabene1111 
with the link to the code I could add additional LDAP logging. For references ..to see more log from Django Auth LDAP add following to 
recipes/settings.py

```
    import logging, logging.handlers
    logfile = "/tmp/django-ldap-debug.log"
    my_logger = logging.getLogger('django_auth_ldap')
    my_logger.setLevel(logging.DEBUG)
    handler = logging.handlers.RotatingFileHandler(
       logfile, maxBytes=1024 * 500, backupCount=5)
    my_logger.addHandler(handler)

```

With this I could trace down the problem to the missing SSL certificate of the ldap server as we use ldaps
As a dirty workaround I added the cert file and the LDAP option into settings.py

```
    LDAP_CA_FILE_PATH = "/opt/recipes/staticfiles/ldapserver.crt"
    AUTH_LDAP_CONNECTION_OPTIONS = {     
    ldap.OPT_X_TLS_CACERTFILE: LDAP_CA_FILE_PATH,
    ldap.OPT_X_TLS_REQUIRE_CERT: ldap.OPT_X_TLS_ALLOW,
    ldap.OPT_X_TLS_NEWCTX: 0
    }
```

and finally got a working connection to the LDAP server.


Kind Regards

**Reply by vabene1111** [reply]
awesome, glad you got it working
